### PR TITLE
address sim2h ws switchout code review comments

### PR DESCRIPTION
**Pull request by neonphog**
- use Option#map instead of match
- make thread panics take down the whole process
- name the threads by the cpu index
- add some extension trait comments

### CODE REVIEW

File: crates/in_stream/src/tcp.rs [hunk 1]
```diff
@@ -182,14 +182,11 @@ impl InStream<&mut [u8], &[u8]> for InStreamTcp {
...
                     addr,
-                    connect_timeout: match config.connect_timeout_ms {
-                        None => None,
-                        Some(ms) => Some(
-                            std::time::Instant::now()
-                                .checked_add(std::time::Duration::from_millis(ms))
-                                .unwrap(),
-                        ),
-                    },
+                    connect_timeout: config.connect_timeout_ms.map(|ms| {
```
Comment (struktured) [inline crates/in_stream/src/tcp.rs]: Wooh functional programming!

File: crates/sim2h/src/job/pool.rs [hunk 2]
```diff
@@ -7,30 +7,50 @@ pub(crate) struct Pool {
...
+            std::process::exit(1);
+        }));
+        true
+    };
+}
+
 impl Pool {
     pub(crate) fn new() -> Self {
+        // make sure if a thread panics, the whole process exits
+        assert!(*SET_THREAD_PANIC_FATAL);
```
Comment (struktured) [inline crates/sim2h/src/job/pool.rs]: Would this get optimized away with a compiler flag? If so, intentional?

File: crates/sim2h/src/job/pool.rs [hunk 3]
```diff
@@ -7,30 +7,50 @@ pub(crate) struct Pool {
...
-                    if job.run() {
-                        send.send(job).expect("failed to send job");
-                    }
-                }
-
-                std::thread::yield_now();
-            }));
+            job_threads.push(
+                std::thread::Builder::new()
+                    .name(format!("sim2h-pool-thread-{}", cpu_index))
```
Comment (struktured) [inline crates/sim2h/src/job/pool.rs]: :+1:

File: crates/sim2h/src/lib.rs [hunk 4]
```diff
@@ -47,17 +47,29 @@ use std::{
...
             .expect("failed to obtain mutex lock")
     }
 }
 
+/// if a channel send fails, it means it is disconnected
+/// this extension trait simplifies panic!ing in that case
+/// in a lot of places, we expect the channel to always be open
+/// and don't have the infrustructure to deal with degenerate cases
+/// this trait makes sending more readable when we want to panic! on disconnects
 pub(crate) trait SendExt<T> {
```
Comment (struktured) [inline crates/sim2h/src/lib.rs]: :+1:

File: crates/sim2h/src/job/pool.rs (hunk 2 above)
Comment (neonphog) [reply crates/sim2h/src/job/pool.rs]: I hope not! - the dereference to trigger the hook update is the whole point : )

**Comment by struktured** [comment]
Just researched it, it's safe:
https://doc.rust-lang.org/std/macro.assert.html

**Pull request merged**